src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -91,21 +140,19 @@ public function __construct($name, $value = null, $expire = 0, $path = '/', $dom
      */
     public function __toString()
     {
-        $str = urlencode($this->getName()).'=';
+        $str = ($this->isRaw() ? $this->getName() : urlencode($this->getName())).'=';
 
         if ('' === (string) $this->getValue()) {
             $str .= 'deleted; expires='.gmdate('D, d-M-Y H:i:s T', time() - 31536001);
         } else {
-            $str .= urlencode($this->getValue());
+            $str .= $this->isRaw() ? $this->getValue() : urlencode($this->getValue());
 
             if ($this->getExpiresTime() !== 0) {
                 $str .= '; expires='.gmdate('D, d-M-Y H:i:s T', $this->getExpiresTime());
             }
         }
 
-        if ($this->path) {
-            $str .= '; path='.$this->path;
-        }
+        $str .= '; path='.$this->getPath() ?: '/';
```

Comment:
Based on https://github.com/ro0NL/symfony/blob/721fc351b2dcf81ba1888682a42155288f5474d2/src/Symfony/Component/HttpFoundation/Cookie.php#L124 i chose to force `/` by default here as well. Maybe it's not needed indeed.

I guess the only case we dont have a path is when a developer overrides `Cookie::$path` (set it manually).

What about
```php
if ($this->getPath()) {
   //..
}
```

to be consistent with the domain attribute handling.